app/models/feed.rb

```diff
@@ -143,7 +143,7 @@ def filtered_clusters(args = {})
     query = self.clusters
 
     # Filter by workspace
-    query = query.where("ARRAY[?] && team_ids", team_ids.to_a.map(&:to_i)) unless team_ids.blank?
+    query = query.where.not("ARRAY[?] && team_ids", self.team_ids - team_ids.to_a.map(&:to_i)) if !team_ids.blank? && team_ids != self.team_ids
```

Comment:
Imagine that the feed is joined by workspaces 1 and 2. Previously if the filter was set to `[1]`, we would return all clusters  that `1` is part of, including the ones that `2` is also part of. Now, if the filter is set to `[1]`. we want to _exclude_ `2` completely, so, only the clusters that `1` is not part of should be returned.